easybuild/easyconfigs/k/KITE/KITE-1.1-gompi-2022a.eb

```diff
@@ -0,0 +1,38 @@
+# Author: J. Sassmannshausen (Imperial College London/UK)
+
+easyblock = 'CMakeMake'
+
+name = 'KITE'
+version = '1.1'
+
+homepage = 'https://github.com/quantum-kite/kite'
+description = """KITE is an open-source Python/C++ software suite for efficient real-space tight-binding (TB) 
+simulations of electronic structure and bulk quantum transport properties of disordered systems scalable to 
+multi billions of atomic orbitals."""
+
+toolchain = {'name': 'gompi', 'version': '2022a'}
+
+source_urls = ['https://github.com/quantum-kite/kite/archive/']
+sources = ['%(version)s.tar.gz']
+
+checksums = [
+    'a480fe46f2220ab6c2d6b9914e0673580121de703523dc22658fe50dc676e47e',   # 1.1.tar.gz
+]
+
+builddependencies = [
+    ('binutils', '2.38'),
+    ('CMake', '3.23.1'),
+]
+
+dependencies = [
+    ('Python', '3.10.4'),
+    ('Eigen', '3.4.0'),
+    ('HDF5', '1.12.2'),
+]
+
+sanity_check_paths = {
+    'files': ['bin/KITEx'],
+    'dirs': [],
+}
```

Comment:
can you add a `sanity_check_commands` here? something like this:
```
sanity_check_commands = ['KITEx |grep "^Starting program..."']
```

Reply:
I tried the usual stuff:
```
$ KITEx -h ; echo $?
[ ... ]
terminate called after throwing an instance of 'H5::FileIException'
Aborted
134
```
The same goes for `$ KITEx -v ; echo $?` which is why I not tried that. 

That is also spitting out quite a lot of messages:
```
$ KITEx -h | grep "Starting program..." &>/dev/null ; echo $?
HDF5-DIAG: Error detected in HDF5 (1.12.2) thread 0:
  #000: H5F.c line 620 in H5Fopen(): unable to open file
    major: File accessibility
    minor: Unable to open file
  #001: H5VLcallback.c line 3501 in H5VL_file_open(): failed to iterate over available VOL connector plugins
    major: Virtual Object Layer
    minor: Iteration failed
  #002: H5PLpath.c line 578 in H5PL__path_table_iterate(): can't iterate over plugins in plugin path '(null)'
    major: Plugin for dynamically loaded library
    minor: Iteration failed
  #003: H5PLpath.c line 620 in H5PL__path_table_iterate_process_path(): can't open directory: /usr/local/hdf5/lib/plugin
    major: Plugin for dynamically loaded library
    minor: Can't open directory or file
  #004: H5VLcallback.c line 3351 in H5VL__file_open(): open failed
    major: Virtual Object Layer
    minor: Can't open object
  #005: H5VLnative_file.c line 97 in H5VL__native_file_open(): unable to open file
    major: File accessibility
    minor: Unable to open file
  #006: H5Fint.c line 1834 in H5F_open(): unable to open file: name = '-h', tent_flags = 0
    major: File accessibility
    minor: Unable to open file
  #007: H5FD.c line 723 in H5FD_open(): open failed
    major: Virtual File Layer
    minor: Unable to initialize object
  #008: H5FDsec2.c line 352 in H5FD__sec2_open(): unable to open file: name = '-h', errno = 2, error message = 'No such file or directory', flags = 0, o_flags = 0
    major: File accessibility
    minor: Unable to open file
terminate called after throwing an instance of 'H5::FileIException'
0
```
So I am not sure if that is of any good, given the error message which is being printed out. When I first noticed it, as in the example above, I basically did not explore more. 
I am happy to implement your suggestion, don't get me wrong here. Part of this is simply for documentation purpose. :-)

Reply:
the error messages are shown because there are no options `-h`, or `-v`.
the program simply expects the second argument to be a file.

the pipe to grep is done just to avoid a failing test, because the program stops with non-zero exit code.

if you don't provide a second argument, there are no errors printed:
```
$ KITEx

+------------------------------------------------------------------------+
|            Chebyshev Polynomial Green's Function Approach              |
|             to Real-Space Quantum Transport Simulations                |
|                                                                        |
|                     KITE | Release 1.1                                 |
|                     Kite home: quantum-kite.com                        |
|                                                                        |
|    Reference: S. M. Joao et al., R. Soc. Open Sci. 7, 191809 (2020)    |
|                                                                        |
|    Created by Simao M. Joao, Joao V. Lopes (Universidade do Porto),    |
|      Tatiana G. Rappoport (Universidade Federal Rio de Janeiro),       |
|        Misa Andelkovic, Lucian Covaci (University of Antwerp)          |
|                and Aires Ferreira (University of York)                 |
|                                                                        |
|            Funded by The Royal Society | royalsociety.org              |
|                                                                        |
|    Copyright (C) 2018-2022, M. Andelkovic, L. Covaci, A. Ferreira,     |
|                S. M. Joao, J. V. Lopes, T. G. Rappoport                |
|                                                                        |
|  This program is free software: you can redistribute it and/or modify  |
|  it under the terms of the GNU General Public License as published by  |
|  the Free Software Foundation, either version 3 of the License, or     |
|  (at your option) any later version.                                   |
|                                                                        |
|  This program is distributed in the hope that it will be useful,       |
|  but WITHOUT ANY WARRANTY; without even the implied warranty of        |
|  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.                  |
|  See the GNU General Public License for more details.                  |
+------------------------------------------------------------------------+

------------------------------ INFORMATION ------------------------------
Linear response functions in units of e^2/h.
To stop these messages, set VERBOSE to 0 in the Makefile.
To see debug messages, set DEBUG to 1 in the Makefile.
-------------------------------------------------------------------------

------- FLAGS SET -------
Flags set at compilation:
DEBUG: 0
VERBOSE: 1
-------------------------

Starting program...

No configuration file found. Exiting.
```

Reply:
I think I need new glasses. Somehow I simply added the `-h` or `-v` as that is what most programs are accepting. 
Entirely my wrong, apologies for that! I have corrected that now. Thanks!

Reply:
thanks!
and no worries, you don't have to apologize, we're all trying to help each other, right :)